sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -134,3 +139,70 @@ object ParserUtils extends SparkParserUtils {
     sb.toString()
   }
 }
+
+object LabelUtils {
+  private val seenLabels: mutable.Set[String] = mutable.Set.empty
+
+  def init(): Unit = {
+    seenLabels.clear()
+  }
+
+  private def checkLabels(
+      beginLabelCtx: Option[BeginLabelContext],
+      endLabelCtx: Option[EndLabelContext]) : Unit = {
+    (beginLabelCtx, endLabelCtx) match {
+      case (Some(bl: BeginLabelContext), Some(el: EndLabelContext))
+        if bl.multipartIdentifier().getText.nonEmpty &&
```

Comment:
If begin label is empty, will the check always pass? Is this intended?

Reply:
This checks that if both labels are defined (not None) and if they hold different string value it should fail as _BeginLabel_ and _EndLabel_ must match.

When I look at it now, I think we can remove `bl.multipartIdentifier().getText.nonEmpty` part from the check because we know that label is defined.

Reply:
I think you are right, but also doesn't hurt to have this check in case someone changes grammar for `multipartIdentifier` in the future for whatever reason.

Reply:
@davidm-db can you please confirm this?

Reply:
Removed my comment already, I wrote it before fully realizing what's going on :)

Reply:
In case you haven't seen my previous comment, I think this looks good at the moment.